{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "import os\n",
    "import pandas as pd\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from datetime import date\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "metadata": {},
   "outputs": [],
   "source": [
    "#EUKaryota genes\n",
    "og_path = '/vortexfs1/omics/alexander/halexander/2020-05-euk-abund/phylogenomics/reference-genomes/busco_downloads/lineages/eukaryota_odb10/info/ogs.id.info'\n",
    "BUSCO_LIST = list(set(list(pd.read_csv(og_path, sep='\\t', header=None)[1])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 393,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/vortexfs1/omics/alexander/halexander/2020-05-euk-abund/metaeuk/busco/eukaryota_odb10']"
      ]
     },
     "execution_count": 393,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#BASE PATH TO MAG BUSCO RUNS\n",
    "BASE_PATH=os.path.join('/vortexfs1/omics/alexander/halexander/2020-05-euk-abund/metaeuk/busco/eukaryota_odb10')\n",
    "                       \n",
    "glob.glob(BASE_PATH)\n",
    "                       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "busco = pd.read_csv('/vortexfs1/omics/alexander/halexander/2020-05-euk-abund/metaeuk/busco-report.tsv', \n",
    "            sep='\\t', index_col=0)\n",
    "busco['Proportion_Complete'] = (busco.Complete + busco.Fragmented)/255\n",
    "busco['Proportion_Complete'] = (busco.Complete) /255\n",
    "complete_bins = busco.loc[busco.Proportion_Complete>=.5].index\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "complete_bins."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loop through identified completed MAGs (>50% BUSCO) and create a table that contains a boolean related to the presence / absence of the protein within the MAG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {},
   "outputs": [],
   "source": [
    "i=complete_bins[0]\n",
    "busco_completeness_df = pd.DataFrame(columns = complete_bins,index=BUSCO_LIST)\n",
    "busco_completeness_df=busco_completeness_df.fillna(0)\n",
    "busco_completeness_fragmented_df=busco_completeness_df.fillna(0)\n",
    "for i in complete_bins:\n",
    "    df=pd.read_csv(os.path.join(BASE_PATH, i, 'run_eukaryota_odb10', \n",
    "                'full_table.tsv'), sep='\\t', header=2, index_col=0)\n",
    "    complete=df.loc[df.Status == 'Complete'].index\n",
    "    fragmented=df.loc[df.Status == 'Fragmented'].index\n",
    "    busco_completeness_df.loc[complete,i]=1\n",
    "    busco_completeness_fragmented_df.loc[fragmented,i]=1\n",
    "    busco_completeness_fragmented_df.loc[complete,i]=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x2aab2d1110f0>"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXAAAAD4CAYAAAD1jb0+AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8li6FKAAASwklEQVR4nO3df4zkdX3H8edbwHiy9oAeTs+rurbiD2QFe1tqtX/s+qOekAZoNZUa5IR2bVOJSTcNF5MqhphcU9EmVWvxRzgb64ZEqJRDW3KyEqot3bMHexQsVC6UkxxFjsMlF9s93v1jvwfr3uzOd3fmOzcffD6Syc73x3znle/svPY73/l+vxuZiSSpPM873gEkSWtjgUtSoSxwSSqUBS5JhbLAJalQJ/bzyTZs2JDDw8NdL+epp57i5JNP7j5Qw0rIWUJGKCOnGXunhJz9zLh79+7HMvP0YyZkZt9umzdvzl647bbberKcppWQs4SMmWXkNGPvlJCznxmBmWzTqe5CkaRCWeCSVCgLXJIKZYFLUqEscEkqlAUuSYXqWOAR8YKIuDMi7oqIeyLiY9X4qyJif0TsqW7nNR9XknRUnRN5fgK8JTPnIuIk4I6I+EY17VOZ+Ynm4kmSltOxwKuDyOeqwZOqmxcRl6TjLLLGP3SIiBOA3cArgc9k5pURcRWwFXgSmAEmM/Ngm8dOABMArVZr89TUVNeh5+bmGBoa6no5TSshZwkZoYycZqxvdv+hFae31sGBw+2njWxa30Ci1evnuhwfH9+dmaNLx9cq8GdmjjgFuBG4Avgf4DEWtsavBjZm5mUrPX50dDRnZmZWk7ut6elpxsbGul5O00rIWUJGKCOnGesb3rZzxemTI/NcM9t+B8G+7ec3EWnV+rkuI6Jtga/qKJTMfAKYBrZk5oHMPJKZTwOfB87tSVJJUi11jkI5vdryJiLWAW8D7ouIjYtmuwjY20xESVI7dY5C2QjsqPaDPw+4PjNvjoi/jYhzWNiFsg/4QHMxJUlL1TkK5W7gDW3GX9JIIklSLZ6JKUmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgplgUtSoSxwSSqUBS5JhbLAJalQFrgkFcoCl6RCWeCSVCgLXJIKZYFLUqEscEkqlAUuSYWywCWpUBa4JBXKApekQnUs8Ih4QUTcGRF3RcQ9EfGxavxpEXFrRNxf/Ty1+biSpKPqbIH/BHhLZp4NnANsiYg3AtuAXZl5BrCrGpYk9UnHAs8Fc9XgSdUtgQuAHdX4HcCFjSSUJLUVmdl5pogTgN3AK4HPZOaVEfFEZp6yaJ6DmXnMbpSImAAmAFqt1uapqamuQ8/NzTE0NNT1cppWQs4SMkIZOc1Y3+z+QytOb62DA4fbTxvZtL6BRKvXz3U5Pj6+OzNHl46vVeDPzBxxCnAjcAVwR50CX2x0dDRnZmbqp17G9PQ0Y2NjXS+naSXkLCEjlJHTjPUNb9u54vTJkXmumT2x7bR9289vItKq9XNdRkTbAl/VUSiZ+QQwDWwBDkTExmrhG4FHe5BTklRTnaNQTq+2vImIdcDbgPuAm4BLq9kuBb7eVEhJ0rHaf0b5aRuBHdV+8OcB12fmzRHxXeD6iLgceAh4d4M5JUlLdCzwzLwbeEOb8T8C3tpEKElSZ56JKUmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgpV5zhwSc9BnU5n1+BzC1ySCmWBS1KhLHBJKpQFLkmFssAlqVAehSIVbLVHkkyOzLPVo0+eM9wCl6RCWeCSVCgLXJIKZYFLUqEscEkqlAUuSYWywCWpUBa4JBWqY4FHxEsj4raIuDci7omID1Xjr4qI/RGxp7qd13xcSdJRdc7EnAcmM/N7EfEiYHdE3FpN+1RmfqK5eJKk5XQs8Mx8BHikuv/jiLgX2NR0MEnSyiIz688cMQzcDpwF/AmwFXgSmGFhK/1gm8dMABMArVZr89TUVLeZmZubY2hoqOvlNK2EnCVkhDJyHo+Ms/sPrWr+1jo4cLihMD20Us6RTev7G2YZ/Xy9x8fHd2fm6NLxtQs8IoaAbwMfz8wbIqIFPAYkcDWwMTMvW2kZo6OjOTMzs+rwS01PTzM2Ntb1cppWQs4SMkIZOY9HxrVczOqa2cG/ht1KOfdtP7/Padrr5+sdEW0LvNZRKBFxEvA14CuZeQNAZh7IzCOZ+TTweeDcXgaWJK2szlEoAXwRuDczP7lo/MZFs10E7O19PEnScup8lnozcAkwGxF7qnEfBi6OiHNY2IWyD/hAIwklSW3VOQrlDiDaTLql93EkSXV5JqYkFcoCl6RCWeCSVCgLXJIKZYFLUqEscEkqlAUuSYWywCWpUBa4JBXKApekQlngklQoC1ySCmWBS1KhLHBJKpQFLkmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgrVscAj4qURcVtE3BsR90TEh6rxp0XErRFxf/Xz1ObjSpKOqrMFPg9MZuZrgTcCfxwRZwLbgF2ZeQawqxqWJPVJxwLPzEcy83vV/R8D9wKbgAuAHdVsO4ALmwopSTpWZGb9mSOGgduBs4CHMvOURdMOZuYxu1EiYgKYAGi1Wpunpqa6jAxzc3MMDQ11vZymlZCzhIxQRs7jkXF2/6FVzd9aBwcONxSmh1bKObJpfX/DLKOfr/f4+PjuzBxdOr52gUfEEPBt4OOZeUNEPFGnwBcbHR3NmZmZVUY/1vT0NGNjY10vp2kl5CwhI5SR83hkHN62c1XzT47Mc83siQ2l6Z2Vcu7bfn6f07TXz9c7ItoWeK2jUCLiJOBrwFcy84Zq9IGI2FhN3wg82quwkqTO6hyFEsAXgXsz85OLJt0EXFrdvxT4eu/jSZKWU+ez1JuBS4DZiNhTjfswsB24PiIuBx4C3t1MRElSOx0LPDPvAGKZyW/tbRxJUl2D/22GVIDhbTuZHJln6yq/VITB+VJO5fFUekkqlAUuSYWywCWpUBa4JBXKApekQnkUinScrfZ0+J913ayv59oRP26BS1KhLHBJKpQFLkmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgplgUtSoSxwSSqUBS5JhbLAJalQFrgkFcoCl6RCdSzwiPhSRDwaEXsXjbsqIvZHxJ7qdl6zMSVJS9XZAr8O2NJm/Kcy85zqdktvY0mSOulY4Jl5O/B4H7JIklYhMrPzTBHDwM2ZeVY1fBWwFXgSmAEmM/PgMo+dACYAWq3W5qmpqa5Dz83NMTQ01PVymlZCzhIywuDnnN1/iNY6OHD4eCdZWQkZobmcI5vW92xZ/fydHB8f352Zo0vHr7XAW8BjQAJXAxsz87JOyxkdHc2ZmZnVJW9jenqasbGxrpfTtBJylpARBj/n8LadTI7Mc83sYP+XwhIyQnM5e/kv1fr5OxkRbQt8TUehZOaBzDySmU8DnwfO7TagJGl11lTgEbFx0eBFwN7l5pUkNaPjZ5SI+CowBmyIiIeBjwJjEXEOC7tQ9gEfaDCjJKmNjgWemRe3Gf3FBrJIklbBMzElqVAWuCQVygKXpEJZ4JJUKAtckgplgUtSoQb/nFr9TBnetrPt+MmRebYuM+2oXp4mLZXALXBJKpQFLkmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgplgUtSoSxwSSqUBS5JhbLAJalQFrgkFcoCl6RCdSzwiPhSRDwaEXsXjTstIm6NiPurn6c2G1OStFSdLfDrgC1Lxm0DdmXmGcCualiS1EcdCzwzbwceXzL6AmBHdX8HcGGPc0mSOljrPvBWZj4CUP18ce8iSZLqiMzsPFPEMHBzZp5VDT+Rmacsmn4wM9vuB4+ICWACoNVqbZ6amuo69NzcHENDQ10vp2kl5By0jLP7D7Ud31oHBw4397wjm9Z39fjZ/Ycaz9gLJWSE5nJ2+zov1s/3zvj4+O7MHF06fq3/Uu1ARGzMzEciYiPw6HIzZua1wLUAo6OjOTY2tsanfNb09DS9WE7TSsg5aBmX+7dpkyPzXDPb3H8A3Pfesa4ev3XbzsYz9kIJGaG5nN2+zosNwntnrbtQbgIure5fCny9N3EkSXXVOYzwq8B3gVdHxMMRcTmwHXh7RNwPvL0aliT1UcfPKJl58TKT3trjLJKkVfBMTEkqlAUuSYWywCWpUBa4JBXKApekQlngklSowT8lq3DDy5xZWMe+7ef3MIlW0s3rpHL08nWeHJlf9szhdpp4P7sFLkmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgplgUtSoSxwSSqUBS5JhbLAJalQFrgkFcproegY3V4vwmu4SP3hFrgkFcoCl6RCdbULJSL2AT8GjgDzmTnai1CSpM56sQ98PDMf68FyJEmr4C4USSpUZObaHxzxIHAQSOBvMvPaNvNMABMArVZr89TU1Jqf76i5uTmGhoa6Xk7T5ubmePDQkTU/fmTT+jU/dnb/oVrztdbBgcNrfpq+KSGnGXunhJyrzdjN+3l8fHx3u13U3Rb4SzLzhxHxYuBW4IrMvH25+UdHR3NmZmbNz3fU9PQ0Y2NjXS+nadPT02z95lNrfnw3h+PVPRRwcmSea2YH/2jSEnKasXdKyLnajN28nyOibYF3tQslM39Y/XwUuBE4t5vlSZLqW3OBR8TJEfGio/eB3wT29iqYJGll3XxGaQE3RsTR5fxdZn6zJ6kkSR2tucAz8wfA2T3MIklahcH+lmBArPXaIJMj83Sziru9Jomk5zaPA5ekQlngklQoC1ySCmWBS1KhLHBJKpQFLkmFssAlqVAWuCQVygKXpEJZ4JJUKAtckgpVzLVQFl8XZHJknq2ruE5INxdSl6RB5Ra4JBXKApekQlngklQoC1ySCmWBS1KhijkKpRv+ZxtJz0VugUtSoSxwSSpUVwUeEVsi4vsR8UBEbOtVKElSZ2su8Ig4AfgM8E7gTODiiDizV8EkSSvrZgv8XOCBzPxBZv4vMAVc0JtYkqROIjPX9sCIdwFbMvP3q+FLgF/LzA8umW8CmKgGXw18f+1xn7EBeKwHy2laCTlLyAhl5DRj75SQs58ZX56Zpy8d2c1hhNFm3DF/DTLzWuDaLp7n2CeOmMnM0V4uswkl5CwhI5SR04y9U0LOQcjYzS6Uh4GXLhr+ReCH3cWRJNXVTYH/G3BGRLwiIp4PvAe4qTexJEmdrHkXSmbOR8QHgX8ETgC+lJn39CzZynq6S6ZBJeQsISOUkdOMvVNCzuOecc1fYkqSji/PxJSkQlngklSogS7wTqfqR8R7I+Lu6vadiDh7ADNeUOXbExEzEfEb/c5YJ+ei+X41Io5Ux/n3VY11ORYRh6p1uSciPtLvjHVyVvOMVRnviYhvD1rGiPjTRetxb/WanzZgGddHxD9ExF3Venx/P/OtIuepEXFj9T6/MyLO6lu4zBzIGwtfjP4X8EvA84G7gDOXzPMm4NTq/juBfx3AjEM8+13D64H7BnFdLprvW8AtwLsGLSMwBtxcwO/lKcB/AC+rhl88aBmXzP9bwLcGLSPwYeDPq/unA48Dzx/AnH8BfLS6/xpgV7/yDfIWeMdT9TPzO5l5sBr8FxaORR+0jHNZvbLAybQ52akP6l724Arga8Cj/QxXKeXSDHVy/h5wQ2Y+BJCZ/V6fq12XFwNf7UuyZ9XJmMCLIiJY2BB6HJjvb8xaOc8EdgFk5n3AcES0+hFukAt8E/Dfi4YfrsYt53LgG40mOlatjBFxUUTcB+wELutTtsU65oyITcBFwOf6mGuxuq/3r1cfqb8REa/rT7SfUifnq4BTI2I6InZHxPv6lm5B7fdORLwQ2MLCH+5+qpPx08BrWThBcBb4UGY+3Z94z6iT8y7gtwEi4lzg5fRpY3KQC7zWqfoAETHOQoFf2WiiNk/dZly7ywncmJmvAS4Erm481bHq5PxL4MrMPNKHPO3Uyfg9Fq4JcTbwV8DfN57qWHVynghsBs4H3gH8WUS8qulgi9R+77Cw++SfM/PxBvO0UyfjO4A9wEuAc4BPR8TPNR1siTo5t7PwB3sPC59i/50+fVIY5H+pVutU/Yh4PfAF4J2Z+aM+ZTtqVZcTyMzbI+KXI2JDZvbzQj11co4CUwufVtkAnBcR85nZr5LsmDEzn1x0/5aI+OyArsuHgccy8yngqYi4HTgb+M/+RFzV7+V76P/uE6iX8f3A9moX5AMR8SAL+5jv7E9EoP7v5fsBqt09D1a35vXzC4FVfnlwIvAD4BU8++XB65bM8zLgAeBNA5zxlTz7JeavAPuPDg9SziXzX0f/v8Sssy5/YdG6PBd4aBDXJQsf+3dV874Q2AucNUgZq/nWs7Bf+eR+rsNVrMe/Bq6q7req986GAcx5CtWXq8AfAF/uV76B3QLPZU7Vj4g/rKZ/DvgI8PPAZ6stx/ns49XBamb8HeB9EfF/wGHgd7N6pQcs53FVM+O7gD+KiHkW1uV7BnFdZua9EfFN4G7gaeALmbl3kDJWs14E/FMufFLoq5oZrwaui4hZFnZlXJn9/bRVN+drgS9HxBEWjj66vF/5PJVekgo1yF9iSpJWYIFLUqEscEkqlAUuSYWywCWpUBa4JBXKApekQv0/WSbsstSken0AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Plot of the relative coverage of protins across buscos\n",
    "(busco_completeness_df.sum(axis=1)/len(busco_completeness_df.columns)).hist(bins=20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 49 of the BUSCO unitigs are present within >80% of the MAGs-- but it is still kind of patchy. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "#80% Protein group list\n",
    "busco_80=busco_completeness_df.loc[(busco_completeness_df.sum(axis=1)/\n",
    "                           len(busco_completeness_df.columns))>=.8].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "#retrieve the instances where a protein is complete or partial \n",
    "Busco_80_test = busco_completeness_fragmented_df.loc[:,(busco_completeness_fragmented_df.loc[busco_80].sum()/49)>.9]\n",
    "Busco_80_test=Busco_80_test.loc[busco_80]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get only the 49 Proteins from all MAGs which have at least 90% of those proteins \n",
    "proteins = Busco_80_test.index\n",
    "MAGs = Busco_80_test.columns\n",
    "ALL_MAGS = complete_bins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['RS-all-SRF-0-8-5-00_bin-105', 'SPO-all-SRF-0-8-5-00_bin-190',\n",
       "       'NAO-all-MES-0-8-5-00_bin-73', 'NAO-all-SRF-0-8-5-00_bin-17',\n",
       "       'SPO-SPSG-MIX-0-8-5-00_bin-132', 'MS-all-DCM-0-8-5-00_bin-224',\n",
       "       'MS-all-DCM-0-8-5-00_bin-636', 'SPO-all-SRF-5-20-00_bin-108',\n",
       "       'SPO-SPSG-DCM-0-8-5-00_bin-172', 'SPO-SPSG-SRF-0-8-5-00_bin-560',\n",
       "       ...\n",
       "       'SPO-SPSG-DCM-5-20-00_bin-40', 'MS-all-SRF-0-8-5-00_bin-123',\n",
       "       'SPO-SPSG-MES-0-8-5-00_bin-179', 'SAO-all-DCM-0-8-5-00_bin-428',\n",
       "       'SPO-SPSG-SRF-0-8-5-00_bin-658', 'SPO-SPSG-SRF-0-8-5-00_bin-277',\n",
       "       'NAO-all-DCM-0-8-5-00_bin-77', 'SAO-all-DCM-0-8-5-00_bin-2',\n",
       "       'SPO-CHIL-DCM-0-8-5-00_bin-46', 'IO-all-SRF-0-8-5-00_bin-89'],\n",
       "      dtype='object', length=234)"
      ]
     },
     "execution_count": 386,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ALL_MAGS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NPO-all-SRF-20-180-00_bin-86      32\n",
       "SAO-all-SRF-0-8-5-00_bin-141      32\n",
       "SAO-all-SRF-20-180-00_bin-71      33\n",
       "NPO-all-SRF-180-2000-00_bin-18    33\n",
       "IO-all-SRF-20-180-00_bin-30       33\n",
       "SAO-all-DCM-0-8-5-00_bin-428      34\n",
       "MS-all-SRF-5-20-00_bin-9          34\n",
       "SAO-all-SRF-0-8-5-00_bin-590      34\n",
       "SAO-all-SRF-180-2000-00_bin-37    34\n",
       "MS-all-DCM-20-180-00_bin-15       34\n",
       "SPO-SPSG-SRF-0-8-5-00_bin-658     35\n",
       "SAO-all-DCM-0-8-5-00_bin-200      35\n",
       "SPO-all-SRF-0-8-5-00_bin-226      36\n",
       "NAO-all-MIX-0-8-5-00_bin-132      36\n",
       "MS-all-SRF-20-180-00_bin-4        37\n",
       "SPO-SPSG-MIX-0-8-5-00_bin-106     37\n",
       "IO-all-SRF-0-8-5-00_bin-126       37\n",
       "SPO-all-SRF-0-8-5-00_bin-551      37\n",
       "RS-all-SRF-0-8-5-00_bin-7         37\n",
       "SAO-all-DCM-0-8-5-00_bin-594      37\n",
       "NAO-all-SRF-0-8-5-00_bin-253      38\n",
       "SPO-SPSG-SRF-0-8-5-00_bin-663     38\n",
       "MS-all-SRF-20-180-00_bin-27       38\n",
       "MS-all-SRF-0-8-5-00_bin-479       38\n",
       "SAO-all-DCM-0-8-5-00_bin-322      38\n",
       "MS-all-SRF-0-8-5-00_bin-131       38\n",
       "IO-all-DCM-0-8-5-00_bin-157       38\n",
       "MS-all-SRF-5-20-00_bin-93         39\n",
       "SPO-SPSG-SRF-0-8-5-00_bin-82      39\n",
       "NPO-all-SRF-0-8-5-00_bin-105      39\n",
       "                                  ..\n",
       "NPO-all-DCM-5-20-00_bin-42        48\n",
       "NAO-all-MES-0-8-5-00_bin-372      48\n",
       "NAO-all-MES-0-8-5-00_bin-73       49\n",
       "MS-all-SRF-0-8-5-00_bin-93        49\n",
       "SPO-SPSG-DCM-0-8-5-00_bin-172     49\n",
       "SAO-all-DCM-0-8-5-00_bin-43       49\n",
       "SAO-all-SRF-0-8-5-00_bin-212      49\n",
       "SAO-all-DCM-0-8-5-00_bin-564      49\n",
       "IO-all-DCM-0-8-5-00_bin-353       49\n",
       "RS-all-SRF-0-8-5-00_bin-17        49\n",
       "SPO-all-SRF-0-8-5-00_bin-867      49\n",
       "SPO-SPSG-SRF-0-8-5-00_bin-608     49\n",
       "SPO-PEOD-DCM-0-8-5-00_bin-41      49\n",
       "NPO-all-SRF-0-8-5-00_bin-232      49\n",
       "NAO-all-MIX-0-8-5-00_bin-96       49\n",
       "SPO-CHIL-SRF-0-8-5-00_bin-39      49\n",
       "IO-all-FSW-0-8-5-00_bin-88        49\n",
       "SAO-all-SRF-0-8-5-00_bin-284      49\n",
       "SPO-all-SRF-0-8-5-00_bin-906      49\n",
       "NAO-all-SRF-0-8-5-00_bin-507      49\n",
       "IO-all-SRF-0-8-5-00_bin-538       49\n",
       "NAO-all-MES-0-8-5-00_bin-268      49\n",
       "SPO-SPSG-DCM-0-8-5-00_bin-173     49\n",
       "NPO-all-SRF-0-8-5-00_bin-407      49\n",
       "IO-all-SRF-0-8-5-00_bin-250       49\n",
       "SAO-all-DCM-0-8-5-00_bin-343      49\n",
       "SAO-all-SRF-0-8-5-00_bin-484      49\n",
       "SPO-all-DCM-0-8-5-00_bin-575      49\n",
       "IO-all-DCM-0-8-5-00_bin-323       49\n",
       "RS-all-SRF-0-8-5-00_bin-105       49\n",
       "Length: 234, dtype: int64"
      ]
     },
     "execution_count": 383,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(busco_completeness_fragmented_df.loc[busco_80].sum()).sort_values()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reference genomes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 394,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reference Genomes BUSCO hits\n",
    "REF_DIR='/vortexfs1/omics/alexander/halexander/2020-05-euk-abund/phylogenomics/reference-genomes/busco/eukaryota_odb10'\n",
    "\n",
    "REF_GENOMES=[]\n",
    "for o in glob.glob(os.path.join(REF_DIR,'*')):\n",
    "    REF_GENOMES.append(os.path.basename(o))\n",
    "\n",
    "REF_busco_completeness_df = pd.DataFrame(columns = REF_GENOMES,index=BUSCO_LIST)\n",
    "REF_busco_completeness_df=REF_busco_completeness_df.fillna(0)\n",
    "REF_busco_completeness_fragmented_df=REF_busco_completeness_df.fillna(0)\n",
    "for i in REF_GENOMES:\n",
    "    df=pd.read_csv(os.path.join(REF_DIR, i, 'run_eukaryota_odb10', \n",
    "                'full_table.tsv'), sep='\\t', header=2, index_col=0)\n",
    "    complete=df.loc[df.Status == 'Complete'].index\n",
    "    fragmented=df.loc[df.Status == 'Fragmented'].index\n",
    "    REF_busco_completeness_df.loc[complete,i]=1\n",
    "    REF_busco_completeness_fragmented_df.loc[fragmented,i]=1\n",
    "    REF_busco_completeness_fragmented_df.loc[complete,i]=1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Partet1        22\n",
       "Brafl1         24\n",
       "Phaglo1        31\n",
       "Symmic1        31\n",
       "Mintr2         32\n",
       "Phaant1        33\n",
       "Oxytri1        34\n",
       "Plafal1        35\n",
       "Dunsal1        35\n",
       "Pavlov2436     36\n",
       "Ochro2298      38\n",
       "Crypto2293     39\n",
       "Blahom1        40\n",
       "Chrsp          41\n",
       "Claok1         42\n",
       "Fracy1         42\n",
       "Toxgon1        42\n",
       "Amoce1         43\n",
       "Auran1         43\n",
       "Pencop1        43\n",
       "Schag1         44\n",
       "Tetthe1        44\n",
       "Chytri1        44\n",
       "Cioin2         44\n",
       "Bigna1         45\n",
       "Physo3         45\n",
       "Naegr1         45\n",
       "Cyccr1         45\n",
       "Vitbras1       45\n",
       "Thaps3         45\n",
       "               ..\n",
       "Nansal1776     46\n",
       "Phyinf1        46\n",
       "Dicpu1         46\n",
       "Psemu1         47\n",
       "Asplep1        47\n",
       "Guith1         47\n",
       "Chlre5         47\n",
       "Capca1         47\n",
       "Aurli1         47\n",
       "MicpuN3v2      47\n",
       "Lotgi1         47\n",
       "Chrpa1         47\n",
       "Aplke1         47\n",
       "Dappu1         47\n",
       "Altfr1         48\n",
       "Ost9901_3      48\n",
       "Phyci1         48\n",
       "Batpra1        48\n",
       "Phatr2         48\n",
       "Pelago2097     48\n",
       "Botmuc1        48\n",
       "Triad1         48\n",
       "MicpuC3v2      48\n",
       "Ostta4221      49\n",
       "Monbr1         49\n",
       "Takru4         49\n",
       "Ascra1         49\n",
       "Claba1         49\n",
       "Helro1         49\n",
       "OstRCC809_2    49\n",
       "Length: 66, dtype: int64"
      ]
     },
     "execution_count": 395,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(REF_busco_completeness_fragmented_df.loc[busco_80].sum()).sort_values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "MAGs = ALL_MAGS\n",
    "\n",
    "for j, o in enumerate(proteins):\n",
    "    all_og=[]\n",
    "    outdir  = os.path.join('extracted', str(date.today()))\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "    og_file = os.path.join('extracted', str(date.today()),o+'.concatenated.faa')\n",
    "    for i, m in enumerate(MAGs):\n",
    "        og_path=''\n",
    "        complete_path = os.path.join(BASE_PATH, m, \n",
    "                'run_eukaryota_odb10','busco_sequences',\n",
    "                'single_copy_busco_sequences', o+'.faa')\n",
    "        frag_path = os.path.join(BASE_PATH, m, \n",
    "                'run_eukaryota_odb10','busco_sequences',\n",
    "                'fragmented_busco_sequences', o+'.faa')\n",
    "        #check for complete OG first\n",
    "        if os.path.isfile(complete_path):\n",
    "            og_path = complete_path\n",
    "        #check for fragmented OG\n",
    "        elif os.path.isfile(frag_path):\n",
    "            og_path = frag_path\n",
    "        #Otherwise just skip it... \n",
    "        if og_path!='':\n",
    "            for record in SeqIO.parse(og_path, \"fasta\"):\n",
    "                record.id = m\n",
    "                record.name=m\n",
    "                all_og.append(record)\n",
    "        #create an empty sequence record with an 'X'-- undetermined amino acid. \n",
    "        else: \n",
    "            record = SeqRecord(Seq(\"-\"),id=m,name=m,description=\"no-record\")\n",
    "            all_og.append(record)\n",
    "    for i, m in enumerate(REF_GENOMES):\n",
    "        og_path=''\n",
    "        complete_path = os.path.join(REF_DIR, m, \n",
    "                'run_eukaryota_odb10','busco_sequences',\n",
    "                'single_copy_busco_sequences', o+'.faa')\n",
    "        frag_path = os.path.join(REF_DIR, m, \n",
    "                'run_eukaryota_odb10','busco_sequences',\n",
    "                'fragmented_busco_sequences', o+'.faa')\n",
    "        #check for complete OG first\n",
    "        if os.path.isfile(complete_path):\n",
    "            og_path = complete_path\n",
    "        #check for fragmented OG\n",
    "        elif os.path.isfile(frag_path):\n",
    "            og_path = frag_path\n",
    "        #Otherwise just skip it... \n",
    "        if og_path!='':\n",
    "            for record in SeqIO.parse(og_path, \"fasta\"):\n",
    "                record.id = m\n",
    "                record.name=m\n",
    "                all_og.append(record)\n",
    "        #create an empty sequence record with an 'X'-- undetermined amino acid. \n",
    "        else: \n",
    "            record = SeqRecord(Seq(\"-\"),id=m,name=m,description=\"no-record\")\n",
    "            all_og.append(record)\n",
    "    with open(og_file, \"w\") as outfasta:\n",
    "        SeqIO.write(all_og, outfasta,'fasta')\n",
    "    with open(os.path.join('extracted', str(date.today()),'ALLMAGS.list'), 'w') as f:\n",
    "        f.write('\\n'.join(MAGs))\n",
    "        f.write('\\n')\n",
    "        f.write('\\n'.join(REF_GENOMES))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.conda-python_lab] *",
   "language": "python",
   "name": "conda-env-.conda-python_lab-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
